### PR TITLE
S04 us23 interface gestion user

### PR DESCRIPTION
**Pull request by RedPandore**

### CODE REVIEW

**Review by jaldabaoth-code — DISMISSED** [review]

**Review by jaldabaoth-code — DISMISSED** [review]

File: src/Controller/UserController.php [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+    {
+        if ($this->isCsrfTokenValid('delete' . $user->getId(), $request->request->get('_token'))) {
+            $entityManager = $this->getDoctrine()->getManager();
+            $entityManager->remove($user);
+            $entityManager->flush();
+        }
+
+        return $this->redirectToRoute('user_index');
+    }
+}
```
Comment (LucyMerlier) [inline src/Controller/UserController.php]: if all your routes in this controller start with "user_", why not put it in your class route annotation?

File: src/Form/UserType.php [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+{
+    public function buildForm(FormBuilderInterface $builder, array $options): void
+    {
+        $builder
+            ->add('email')
+            ->add('roles')
+            ->add('password')
+            ->add('createdAt')
+            ->add('username')
+            ->add('registeredUser')
```
Comment (LucyMerlier) [inline src/Form/UserType.php]: isn't this supposed to be an EntityType?

File: src/Form/UserType.php [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+use Symfony\Component\Form\FormBuilderInterface;
+use Symfony\Component\OptionsResolver\OptionsResolver;
+
+class UserType extends AbstractType
+{
+    public function buildForm(FormBuilderInterface $builder, array $options): void
+    {
+        $builder
+            ->add('email')
+            ->add('roles')
```
Comment (LucyMerlier) [inline src/Form/UserType.php]: isn't this supposed to be a ChoiceType?

File: src/Form/UserType.php [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+            ->add('password')
+            ->add('createdAt')
```
Comment (LucyMerlier) [inline src/Form/UserType.php]: isn't this supposed to be auto-generated, not inputed?

File: templates/user/index.html.twig [hunk 5]
```diff
@@ -0,0 +1,97 @@
...
+                                {% endif %}
+
+                                {% if user.roles == ["ROLE_SUPERADMIN",
+                                                        "ROLE_USER"] %}
+
+                                    {% elseif user.roles == ["ROLE_ADMIN",
+                                                                "ROLE_USER"] and app.user.roles == ["ROLE_ADMIN",
+                                                                                                        "ROLE_USER"] %}
+
+                                    {% else %}
```
Comment (LucyMerlier) [inline templates/user/index.html.twig:78]: so this if and this elseif don't do anything?

File: templates/user/index.html.twig [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+
+                                {% endif %}
+
+                            </td>
+                        </tr>
+
+                    {% else %}
+
+                        <tr>
+                            <td colspan="7">Il n'y à aucun utilisateur inscrit.</td>
```
Comment (LucyMerlier) [inline templates/user/index.html.twig]: "a" => not "à"

**Review by LucyMerlier — CHANGES_REQUESTED** [review]
see comments

**Review by bouboumael — APPROVED** [review]

**Review by LucyMerlier — APPROVED** [review]

**Pull request merged**